cmd/state/main.go

```diff
@@ -150,6 +150,18 @@ func run(args []string, isInteractive bool, cfg *config.Instance, out output.Out
 
 	svcmodel := model.NewSvcModel(svcPort)
 
+	// Give the logger a way to fetch the state-svc log tail. This cannot be done inside the logger
+	// package itself because importing pkg/platform/model creates an import cycle.
+	logging.SetSvcTailProvider(func() string {
+		ctx, cancel := context.WithTimeout(context.Background(), model.SvcTimeoutMinimal)
+		defer cancel()
+		tail, err := svcmodel.FetchLogTail(ctx)
+		if err != nil {
+			return fmt.Sprintf("Could not fetch state-svc log: %v", err)
+		}
+		return tail
+	})
```

Comment:
I don't think there's merit in linking "give me the svc log" with the logging library. ie. when we report to rollbar we can call the model function to retrieve the svc log then and there, it doesn't need to be routed through the logging lib itself.

To put it differently; the responsibility of the logging lib is logging of the current runtime. Not providing access to logs of other runtimes.

Reply:
Perhaps what we can do instead is introduce a mechanic to the rollbar library that allows you to amend the logged data before it is sent.

Reply:
That's reasonable. I've added such a function to the rollbar library and utilize it within cmd/state/main.go. (Keep in mind there's still an import cycle trying to use the svcModel within the Rollbar package).